YoCode/CommandLineArguments/CommandExtractor.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace YoCode
+{
+    class CommandExtractor
+    {
+        public static SplitArg ArgsSplitter(string arg)
+        {
+            return new SplitArg
+            {
+                command = (ExtractSuffix(arg,CommandIdentifiers.dataIdentifier) != null) 
+                ? ExtractPrefix(arg.Substring(0, arg.IndexOf(CommandIdentifiers.dataIdentifier)), CommandIdentifiers.commandIdentifier) 
+                : ExtractPrefix(arg, CommandIdentifiers.commandIdentifier),
+
```

Comment:
I _think_ this could be:
```
var argParts = arg.Split(new[] {CommandIdentifiers.commandIdentifier, CommandIdentifiers.dataIdentifier}, StringSplitOptions.RemoveEmptyEntries);
return new SplitArg
{
   command = argParts[0],
   data = argParts[1]
};
```
Check it though, I might have it slightly wrong. Don't just take my word for it. ;)